ksqldb-engine/src/main/java/io/confluent/ksql/engine/QueryIdUtil.java

```diff
@@ -100,7 +100,9 @@ static QueryId buildId(
     }
 
     if (!outputNode.getSinkName().isPresent()) {
-      return new QueryId(String.valueOf(Math.abs(ThreadLocalRandom.current().nextLong())));
+      final String prefix =
+          "transient_" + outputNode.getSource().getLeftmostSourceNode().getAlias().text() + "_";
```

Comment:
there doesn't seem to be any tests that run through this part of the code - can you add tests for that (at least unit tests for `QueryIdUtil` that mock out `outputNode`, but ideally also some end-to-end tests)

Reply:
I added a unit test in `QueryIdUtil ` that checks we've added the source name as desired. However, I have been struggling to find some end-to-end tests that I can follow to help me create one here, are there any you would recommend?

Reply:
You can use `KsqlEngineTest` if you want - otherwise I'm actually OK waiting for when we do #5868 because we'll have to have more testing that covers this for that anyway.

Reply:
is `getLeftmostSourceNode` guaranteed to always be present?  I assume it is, being that it seems it's either filled out in the case of a non-join or the left-most in the case of a join.  Do we do a similar naming convention with persistent queries?